Failed e2e test does not fail CI workflow

According to logs e2e test fails by CI workflow is green:

http://argo.applatix.net/workflows/default/argo-cd-ci-pgx8c?tab=workflow&nodeId=argo-cd-ci-pgx8c-3310564159&sidePanel=logs%3Aargo-cd-ci-pgx8c-3310564159%3Amain


```
time="2018-08-02T14:55:06Z" level=info msg="finished unary call with code OK" grpc.code=OK grpc.method=GenerateManifest grpc.service=repository.RepositoryService grpc.start_time="2018-08-02T14:55:06Z" grpc.time_ms=20.464 span.kind=server system=grpc
--- FAIL: TestAppManagement (268.13s)
   --- PASS: TestAppManagement/TestAppCreation (8.96s)
   --- PASS: TestAppManagement/TestAppDeletion (10.33s)
   --- PASS: TestAppManagement/TestTrackAppStateAndSyncApp (18.85s)
   --- PASS: TestAppManagement/TestAppRollbackSuccessful (17.88s)
   --- FAIL: TestAppManagement/TestComparisonFailsIfClusterNotAdded (180.01s)
        fixture.go:366: Failed to wait for expected condition
   --- PASS: TestAppManagement/TestArgoCDWaitEnsureAppIsNotCrashing (32.09s)
```

--- a/test/e2e/app_management_test.go
+++ b/test/e2e/app_management_test.go
@@ -170,7 +170,7 @@ func TestAppManagement(t *testing.T) {
 
 		WaitUntil(t, func() (done bool, err error) {
 			app, err := fixture.AppClient.ArgoprojV1alpha1().Applications(fixture.Namespace).Get(app.ObjectMeta.Name, metav1.GetOptions{})
-			return err == nil && app.Status.ComparisonResult.Status != v1alpha1.ComparisonStatusUnknown && len(app.Status.Conditions) > 0, err
+			return err == nil && app.Status.ComparisonResult.Status == v1alpha1.ComparisonStatusUnknown && len(app.Status.Conditions) > 0, err
 		})
 
 		app, err := fixture.AppClient.ArgoprojV1alpha1().Applications(fixture.Namespace).Get(app.ObjectMeta.Name, metav1.GetOptions{})

--- a/test/e2e/main_test.go
+++ b/test/e2e/main_test.go
@@ -17,7 +17,8 @@ func TestMain(m *testing.M) {
 		println(fmt.Sprintf("Unable to create e2e fixture: %v", err))
 		os.Exit(-1)
 	} else {
-		defer fixture.TearDown()
-		m.Run()
+		code := m.Run()
+		fixture.TearDown()
+		os.Exit(code)
 	}
 }

